### PR TITLE
Update required `icub_firmware_shared_VERSION` to 1.31.0 for AksIM2 encoder

### PR DESCRIPTION
**Pull request by sgiraz**
**What's new:**
- In order to see the new diagnostics messages generated by the AksIM2 encoder we have to use `icub-firmware-shared` v1.31.0 which contains the new (human-readable) diagnostics messages to visualize on the `yarprobotinterface`

**Notes:**

### CODE REVIEW

**Comment by pattacini** [comment]
Depends on:
- https://github.com/robotology/icub-firmware-shared/pull/77

**Review by pattacini — APPROVED** [review]
Fine with me 👍🏻 

@sgiraz, remember to make the PR ready for review once you have merged the PR in icub-fw-shared.
Then, I'll wait for the CI to complete.

**Review by marcoaccame — APPROVED** [review]
seems fine to me. i will merge the associated PRs on icub-firmware-shared and incub-firmware first.

**Comment by pattacini** [comment]
PR made ready for review 👍🏻 
CI commenced.

**Pull request merged**
